### ISSUE TITLE
missing warning on loosing precision to bit fields 

### ISSUE THREAD

**Issue by wheatman** [llvm/llvm-project#74219]
The following code gives warnings about loosing precision when we use an int to initilize a smaller type
```
class A {
    int i;
    short s;
    char c;
    A(int x, int y, int z) : 
        i(x),
        s(y), // warning: implicit conversion loses integer precision: 'int' to 'short' [-Wimplicit-int-conversion]
        c(z) // warning: implicit conversion loses integer precision: 'int' to 'char' [-Wimplicit-int-conversion]
       {}
};
```

However, if that smaller type is a bitfield as in 
```
class B {
    int i;
    int s : 16;
    int c : 8;
    B(int x, int y, int z) : i(x), s(y), c(z) {}
};
```
no warnings are given

https://godbolt.org/z/7oqY78T58

This issue led to the bug in https://github.com/llvm/llvm-project/issues/71888

**Comment by shafik** [comment]
Looks like this is know issue based on the `TODO` here:

https://github.com/llvm/llvm-project/blob/21a9c7e6e7b89b2000e2249a5305cf5d0b04ccd2/clang/lib/Sema/SemaChecking.cpp#L15753-L15784

**Comment by shafik** [comment]
It looks like the last time we have access to the `FieldDecl` we are initializing is here:

https://github.com/llvm/llvm-project/blob/21a9c7e6e7b89b2000e2249a5305cf5d0b04ccd2/clang/lib/Sema/SemaDeclCXX.cpp#L4670-L4677

at this point we can call `DirectMember->getBitWidthValue(Context)` I don't see a way of getting at this from there to here:

https://github.com/llvm/llvm-project/blob/21a9c7e6e7b89b2000e2249a5305cf5d0b04ccd2/clang/lib/Sema/SemaChecking.cpp#L16012-L16016

where the call to `CheckImplicitConversion(...)` is decided and made.

CC @AaronBallman @erichkeane
